src/test/store/symbolication.test.js

```diff
@@ -0,0 +1,187 @@
+        '- 0x000a (total: 1, self: —)',
+        '  - 0x2000 (total: 1, self: 1)',
+        '- 0x0000 (total: 1, self: —)',
+        '  - 0x2000 (total: 1, self: 1)',
+        '- 0x1a0f (total: 1, self: 1)',
+        '- 0x0f0f (total: 1, self: 1)',
+      ]);
+
+      await doSymbolicateProfile(dispatch, profile, symbolStore);
+      expect(formatTree(getCallTree(getState()))).toEqual([
+        // 0x0000 and 0x000a get merged together.
+        '- first symbol (total: 2, self: —)',
+        '  - last symbol (total: 2, self: 2)',
+        '- third symbol (total: 1, self: 1)',
+        '- second symbol (total: 1, self: 1)',
+      ]);
+    });
+  });
+
+  describe('merging of functions with different memory addresses, but in the same function', () => {
+    it('starts with expanded call nodes of multiple memory addresses', async function() {
+      const { store: { dispatch, getState }, namesAsFuncIndexes } = init();
+
+      const threadIndex = 0;
+      const selectedCallNodePath = namesAsFuncIndexes(['0x000a', '0x2000']);
+      // Both of these expanded nodes are actually in the same function, but
+      // they are different memory addresses.
+      const expandedCallNodePaths = [['0x000a'], ['0x0000']].map(
+        namesAsFuncIndexes
+      );
+
+      dispatch(changeSelectedCallNode(threadIndex, selectedCallNodePath));
+      dispatch(changeExpandedCallNodes(threadIndex, expandedCallNodePaths));
+
+      expect(getSelectedCallNodePath(getState())).toEqual(selectedCallNodePath);
+      expect(getExpandedCallNodePaths(getState())).toEqual(
+        expandedCallNodePaths
+      );
+    });
+
+    it('symbolicates and merges functions in the stored call node paths', async function() {
+      const {
+        store: { dispatch, getState },
+        profile,
+        symbolStore,
+        namesAsFuncIndexes,
+      } = init();
+
+      const threadIndex = 0;
+
+      dispatch(
+        changeSelectedCallNode(
+          threadIndex,
+          namesAsFuncIndexes(['0x000a', '0x2000'])
+        )
+      );
+      // Both of these expanded nodes are actually in the same function, but
+      // they are different memory addresses. See exampleSymbolTable and
+      // _createUnsymbolicatedProfile().
+      dispatch(
+        changeExpandedCallNodes(
+          threadIndex,
+          [['0x000a'], ['0x0000']].map(namesAsFuncIndexes)
+        )
+      );
+
+      await doSymbolicateProfile(dispatch, profile, symbolStore);
+      expect(getSelectedCallNodePath(getState())).toEqual(
+        // The CallNodePath is now symbolicated.
+        namesAsFuncIndexes(['first symbol', 'last symbol'])
+      );
+
+      expect(getExpandedCallNodePaths(getState())).toEqual(
+        // Notice how these are duplicated, however they are equivalent.
+        // See: https://github.com/devtools-html/perf.html/issues/270
+        [['first symbol'], ['first symbol']].map(namesAsFuncIndexes)
+      );
+    });
+  });
```

Comment:
Could you please add a simple test for the symbolicationStatus ? This could be as simple as:
```javascript
const symbolicationPromise = doSymbolicateProfile(...);
expect(getSymbolicationStatus(...), 'SYMBOLICATING');
await symbolicationPromise;
expect(getSymbolicationStatus(...), 'DONE');
```

but you can also use `observeStoreStateChanges` from `test/store/receive-profile.test.js` if you want something that could be more robust (for example if the "startSymbolicating" dispatch is done asynchronously).